api/users/serializers.py

```diff
@@ -117,6 +122,21 @@ def get_absolute_url(self, obj):
             'version': self.context['request'].parser_context['kwargs']['version']
         })
 
+    def get_node_count(self, obj):
+        return obj.nodes.count()
+
+    def get_quickfiles_count(self, obj):
+        return QuickFilesNode.objects.get(contributor__user__id=obj.id).files.filter(type='osf.osfstoragefile').count()
+
+    def get_registration_count(self, obj):
+        return Registration.objects.filter(contributor__user__id=obj.id).count()
+
+    def get_preprint_count(self, obj):
+        return PreprintService.objects.filter(node___contributors__guids___id=obj._id).count()
```

Comment:
The definition for whether or not a user can see a preprint is more complicated. You can look at the preprints_queryset on the `PreprintFilterMixin` to see how the query is built for UserPreprints.   If the preprint is not published or public, you cannot see it unless you are an admin.